packages/api-server/ros_tortoise/__main__.py

```diff
@@ -138,18 +139,24 @@ def main():
     args = parser.parse_args()
     packages = [
         k
-        for k in ament_index_python.get_resources("rosidl_interfaces").keys()
+        for k in ament_index_python.get_resources("rosidl_interfaces")
         if k in args.packages
     ]
     os.makedirs(args.output, exist_ok=True)
+    sha1 = hashlib.sha1()
     for pkg in packages:
         pkg_spec = PackageSpec(pkg, parse_package(pkg))
         mixins = gen_mixin(pkg_spec)
+        sha1.update(mixins.encode())
         outfile = f"{args.output}/{pkg}_mixins.py"
         with open(outfile, "w") as f:
             f.write(mixins)
         print(outfile)
 
+    rev = base64.b32encode(sha1.digest())
+    with open(f"{args.output}/rev", "bw") as f:
+        f.write(rev)
```

Comment:
On models migration to the database, on prod, should we save the rev's hash in case we want to replicate models migration on the localhost, right?  

Is that the idea?

Reply:
The initial plan was to provide an easy way for build system to do change detection, but api client is not using a build system now so it is not used.

The rev only tracks and the mixins and not the models so we cant use it for any prod work.